tianshou/env/vecenv/shmemenv.py

```diff
@@ -0,0 +1,158 @@
+import gym
+import ctypes
+import numpy as np
+from collections import OrderedDict
+from multiprocessing import Pipe, Process, Array
+from typing import Callable, List, Optional, Tuple, Union
+
+
+from tianshou.env import BaseVectorEnv, SubprocVectorEnv
+from tianshou.env.utils import CloudpickleWrapper
+
+_NP_TO_CT = {np.float64: ctypes.c_double,
+             np.float32: ctypes.c_float,
+             np.int32: ctypes.c_int32,
+             np.int8: ctypes.c_int8,
+             np.uint8: ctypes.c_char,
+             np.bool: ctypes.c_bool}
+
+
+def _shmem_worker(parent, p, env_fn_wrapper, obs_bufs,
+                  obs_shapes, obs_dtypes, keys):
+    """Control a single environment instance using IPC and shared memory."""
+    def _encode_obs(maybe_dict):
+        flatdict = maybe_dict if isinstance(maybe_dict, dict) else {
+            None: maybe_dict}
+        for k in keys:
+            dst = obs_bufs[k].get_obj()
+            dst_np = np.frombuffer(
+                dst, dtype=obs_dtypes[k]).reshape(obs_shapes[k])
+            np.copyto(dst_np, flatdict[k])
+        return None
```

Comment:
return None is not necessary.

Reply:
Yes, but I intentionally do this for clarity. A function with no return can sometimes be confusing, especially when its "return" serves as the input of another function for people who are not that familiar with the language.
```
        obs = np.stack(
            [self._decode_obs(self.parent_remote[i].recv(), i) for i in id])
```